web/modules/custom_weather_module/src/Plugin/Block/CustomWeatherBlock.php

```diff
@@ -89,9 +100,52 @@ private function userWeather() {
     }
     $location = $this->userLocation();
     $client = new Client();
-    $request = $client->get('http://api.weatherapi.com/v1/current.json?key=' . $api_key . '&q=' . $location);
-    $response = json_decode($request->getBody(), TRUE);
-    return $response['current']['temp_c'] . ' °C, ' . $response['current']['condition']['text'];
+    try {
+      $request = $client->get('http://api.weatherapi.com/v1/current.json?key=' . $api_key . '&q=' . $location);
+      $response = json_decode($request->getBody(), TRUE);
+      return [
+        'conditions' => $response['current']['temp_c'] . ' °C, ' . $response['current']['condition']['text'],
+        'icon' => $response['current']['condition']['icon'],
+      ];
+    }
+    catch (RequestException $e) {
+      // @todo use dependency injection
+      // phpcs:ignore
+      \Drupal::logger('custom_weather_module')->error('Message from @module: @message', [
+        '@module' => 'custom_weather_module',
+        '@message' => "GuzzleHttp\Exception\ConnectException: cURL error 6: Could not resolve host: weatherapi.com",
+      ]);
```

Comment:
Do not forget that your method always should return the value to easy handling the result of work.